substrate/primitives/core/src/bandersnatch.rs

```diff
@@ -170,42 +189,30 @@ impl CryptoType for Pair {
 /// Bandersnatch VRF types and operations.
 pub mod vrf {
 	use super::*;
-	use crate::{bounded::BoundedVec, crypto::VrfCrypto, ConstU32};
-	use bandersnatch_vrfs::{
-		CanonicalDeserialize, CanonicalSerialize, IntoVrfInput, Message, PublicKey,
-		ThinVrfSignature, Transcript,
-	};
+	use crate::crypto::VrfCrypto;
 
-	/// Max number of inputs/pre-outputs which can be handled by the VRF signing procedures.
-	///
-	/// The number is quite arbitrary and chosen to fulfill the use cases found so far.
-	/// If required it can be extended in the future.
-	pub const MAX_VRF_IOS: u32 = 3;
-
-	/// Bounded vector used for VRF inputs and pre-outputs.
-	///
-	/// Can contain at most [`MAX_VRF_IOS`] elements.
-	pub type VrfIosVec<T> = BoundedVec<T, ConstU32<MAX_VRF_IOS>>;
+	/// [`VrfSignature`] serialized size.
+	pub const VRF_SIGNATURE_SERIALIZED_SIZE: usize =
+		PREOUT_SERIALIZED_SIZE + SIGNATURE_SERIALIZED_SIZE;
 
 	/// VRF input to construct a [`VrfPreOutput`] instance and embeddable in [`VrfSignData`].
 	#[derive(Clone, Debug)]
-	pub struct VrfInput(pub(super) bandersnatch_vrfs::VrfInput);
+	pub struct VrfInput(pub(super) bandersnatch::Input);
 
 	impl VrfInput {
 		/// Construct a new VRF input.
-		pub fn new(domain: impl AsRef<[u8]>, data: impl AsRef<[u8]>) -> Self {
-			let msg = Message { domain: domain.as_ref(), message: data.as_ref() };
-			VrfInput(msg.into_vrf_input())
+		///
+		/// Hash to Curve (H2C) using Elligator2.
+		pub fn new(data: &[u8]) -> Self {
+			Self(bandersnatch::Input::new(data).expect("H2C for Bandersnatch can't fail; qed"))
```

Comment:
Semantically, if it cant fail then it should not return a result.

Reply:
This `Input` constructor relies on a trait implementation for hash-to-curve. The trait method returns a result since failure is possible in general, depending on the specific implementation. However, the built-in Bandersnatch suite in `ark-vrf` is infallible